### PR TITLE
Scottx611x/delete related analysis objects

### LINKED ISSUE #1428: Delete related objects when analysis is deleted
- Specific code commit: 738eba5c
- Version of the web browser and OS: N/A
- Environment where the error occurred (Vagrant VM and site conf mode or AWS instance): Vagrant VM, dev mode
### Steps to reproduce
1. Import a Galaxy workflow and a data set
2. Run the workflow on the data set
3. Delete the resulting analysis in the Django admin (not a bulk delete)
### Observed behavior

Related objects like WorkflowDataInputMap, WorkflowFilesDL (others?) are not deleted
### Expected behavior

Related objects should be deleted

### PR DESCRIPTION
**Pull request by scottx611x**
Resolves #1428 
Towards #1187

### CODE REVIEW

**Comment by scottx611x** [comment]
@hackdna I've made the changes you've requested, and I still feel wrong about it. I already made errors copying and pasting all of theses assertions with my local edits and I would think someone else extending these test cases in the future could easily do the same. 

What if we just avoided your point of unnecessarily retesting of code by calling these `_assert_related_objects*` in a few less places, but kept the repetitive assertions in their own methods still?

File: refinery/core/tests.py [hunk 1]
```diff
@@ -679,6 +699,71 @@ def test_pre_isa_archive_deletion(self):
...
+                1,
+                self.user
+            )
+        self.analysis = self.analyses[0]
+
+        self.assertGreater(Analysis.objects.count(), 0)
+        self.assertGreater(AnalysisNodeConnection.objects.count(), 0)
+        self.assertGreater(AnalysisResult.objects.count(), 0)
+        self.assertGreater(AnalysisStatus.objects.count(), 0)
+        self.assertGreater(Node.objects.count(), 0)
```
Comment (hackdna) [inline refinery/core/tests.py]: No testing in setUp(), move these five lines into a test.

File: refinery/core/tests.py [hunk 2]
```diff
@@ -679,6 +699,71 @@ def test_pre_isa_archive_deletion(self):
...
 
+class AnalysisDeletionTest(TestCase):
+    """Testing for the deletion of Analyses"""
+    def setUp(self):
+        self.username = self.password = 'user'
+        self.user = User.objects.create_user(
+            self.username, '', self.password
+        )
+        self.analyses, self.dataset = \
+            make_analyses_with_single_dataset(
```
Comment (hackdna) [inline refinery/core/tests.py]: This call should fit on a single line

**Comment by hackdna** [comment]
I think there is minimal code duplication and the test cases are easier to read now.

**Pull request merged**